components/FeatureHighlight/src/private/MDCFeatureHighlightAnimationController.m

```diff
@@ -66,26 +67,38 @@ - (void)animateTransition:(id<UIViewControllerContextTransitioning>)transitionCo
         break;
     }
   }
-  [UIView animateWithDuration:transitionDuration
-      delay:0.0
-      options:UIViewAnimationOptionBeginFromCurrentState
-      animations:^{
-        // We have to perform an animation on highlightView in this block or else UIKit
-        // will not know we are performing an animation and will call the completion block
-        // immediately, causing our CAAnimations to be cut short.
-        if (self.presenting) {
-          [highlightView layoutAppearing];
-        } else {
-          [highlightView layoutDisappearing];
-        }
-      }
-      completion:^(__unused BOOL finished) {
-        // If we're dismissing, remove the highlight view from the hierarchy
-        if (!self.presenting) {
-          [fromView removeFromSuperview];
-        }
-        [transitionContext completeTransition:YES];
-      }];
+
+  BOOL presenting = self.presenting;
+  void (^animations)(void) = ^void(void) {
```

Comment:
To avoid context switching, if your block is short, consider placing it inline, as it was before this change.
Capturing BOOL presenting outside of the blocks to avoid referencing self is probably a good idea.

Reply:
I figured that because the code in both the `animations` and `completion` blocks could be called in two different places depending on whether the app was backgrounded or not it made sense to DRY them out into block variables.